test/integration/setup.js

```diff
@@ -1,128 +1,113 @@
+    const implementation = await contractFactory.deploy(...deployParams);
+    const proxy = await OwnedUpgradeabilityProxy.deploy(implementation.address);
+    await proxy.deployed();
+    return await ethers.getContractAt(contract, proxy.address);
   };
 
-  const upgradeProxy = async (proxyAddress, contract, params = []) => {
-    const implementation = await contract.new(...params);
-    const proxy = await OwnedUpgradeabilityProxy.at(proxyAddress);
-    await proxy.upgradeTo(implementation.address);
+  const upgradeProxy = async (proxyAddress, contract, constructorArgs = [], overrides = {}) => {
+    const contractFactory = await ethers.getContractFactory(contract, overrides);
+
+    const impl = await deployImmutable(contractFactory, constructorArgs);
+    const proxy = await ethers.getContractAt('OwnedUpgradeabilityProxy', proxyAddress);
+    await proxy.upgradeTo(impl.address);
+    const instance = await ethers.getContractAt(contract, proxyAddress);
+
+    return instance;
   };
 
   const transferProxyOwnership = async (proxyAddress, newOwner) => {
-    const proxy = await OwnedUpgradeabilityProxy.at(proxyAddress);
+    const proxy = await ethers.getContractAt('OwnedUpgradeabilityProxy', proxyAddress);
     await proxy.transferProxyOwnership(newOwner);
   };
 
-  const [owner, emergencyAdmin] = accounts;
+  const coverUtilsLib = await CoverUtilsLib.deploy();
+  await coverUtilsLib.deployed();
+  const options = { libraries: { CoverUtilsLib: coverUtilsLib.address } };
+
+  const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000';
+  const QE = '0x51042c4d8936a7764d18370a6a0762b860bb8e07';
+  const INITIAL_SUPPLY = parseEther('15000000000');
+
+  const {
+    defaultSender: { address: owner },
+    emergencyAdmin,
+  } = ethersAccounts;
 
   // deploy external contracts
-  const weth = await WETH9.new();
+  const weth = await WETH9.deploy();
+  await weth.deployed();
+
+  const dai = await ERC20BlacklistableMock.deploy();
+  await dai.deployed();
 
-  const dai = await ERC20BlacklistableMock.new();
-  await dai.mint(owner, ether('10000000'));
+  await dai.mint(owner, parseEther('10000000'));
 
-  const stETH = await ERC20BlacklistableMock.new();
-  await stETH.mint(owner, ether('10000000'));
+  const stETH = await ERC20BlacklistableMock.deploy();
+  await stETH.deployed();
+  await stETH.mint(owner, parseEther('10000000'));
 
-  const chainlinkDAI = await ChainlinkAggregatorMock.new();
-  const chainlinkSteth = await ChainlinkAggregatorMock.new();
-  await chainlinkSteth.setLatestAnswer(new BN((1e18).toString()));
-  const priceFeedOracle = await PriceFeedOracle.new(
+  const chainlinkDAI = await ChainlinkAggregatorMock.deploy();
+
+  const chainlinkSteth = await ChainlinkAggregatorMock.deploy();
+  await chainlinkSteth.setLatestAnswer(parseEther('1').toString());
+
+  await chainlinkDAI.deployed();
+  await chainlinkSteth.deployed();
+
+  const priceFeedOracle = await PriceFeedOracle.deploy(
```

Comment:
for example here await priceFeedOracle.deployed() is not used.

and works.

I expect the initial .deploy() is sufficient